traffic_ops/traffic_ops_golang/cachegroupparameter/parameters.go

```diff
@@ -180,3 +185,120 @@ func (cgparam *TOCacheGroupParameter) Delete() (error, error, int) {
 
 	return api.GenericDelete(cgparam)
 }
+
+func ReadAllCacheGroupParameters(w http.ResponseWriter, r *http.Request) {
+	inf, userErr, sysErr, errCode := api.NewInfo(r, nil, nil)
+	if userErr != nil || sysErr != nil {
+		api.HandleErr(w, r, inf.Tx.Tx, errCode, userErr, sysErr)
+		return
+	}
+	defer inf.Close()
+	output, err := GetAllCacheGroupParameters(inf.Tx.Tx)
+	if err != nil {
+		api.WriteRespAlertObj(w, r, tc.ErrorLevel, "querying cachegroupparameters with error: "+err.Error(), output)
+		return
+	}
+	api.WriteResp(w, r, output)
+}
+
+func GetAllCacheGroupParameters(tx *sql.Tx) (tc.CacheGroupParametersList, error) {
+	parameters := map[string]string{
+		"orderby": "cachegroup",
+	}
+	// Query Parameters to Database Query column mappings
+	// see the fields mapped in the SQL query
+	queryParamsToQueryCols := map[string]dbhelpers.WhereColumnInfo{
+		"cachegroup":  dbhelpers.WhereColumnInfo{"cachegroup", api.IsInt},
+		"lastUpdated": dbhelpers.WhereColumnInfo{"last_updated", nil},
+		"parameter":   dbhelpers.WhereColumnInfo{"parameter", api.IsInt},
```

Comment:
this needs to be added to the doc as support request query parameters

Reply:
on second thought i dont think we want to do this. ill take it out